### ISSUE TITLE
Importing the provider and using it leads to a crash

### ISSUE THREAD

**Issue by Josh-Ring-SD** [saraelsa/msal-community-solid#1]
Importing the package and attempting to use it crashes the dev environment from `npm run dev` with Vite, see `package.json` at the bottom.

Also just importing the package causes the exact same crash

```javascript

import { Router, Route, Routes } from "@solidjs/router"
import { Configuration, IPublicClientApplication, PublicClientApplication } from "@azure/msal-browser"
import { msalConfig } from "src/authConfig"
import { MsalProvider, AuthenticatedTemplate, UnauthenticatedTemplate } from "msal-community-solid"

const msalInstance: IPublicClientApplication = new PublicClientApplication(msalConfig)

function ComponentHere() {
	return (<div>testing</div>)
}

function Login() {
	return (<div>login</div>)
}


render(() => (
	<MsalProvider instance={msalInstance} >
		<Router>

			<AuthenticatedTemplate>
				<Routes>
					<Route path="/" component={ComponentHere} />
				</Routes>
			</AuthenticatedTemplate>

			<UnauthenticatedTemplate>
				<Routes>
					<Route path="/login" component={Login} />
				</Routes>
			</UnauthenticatedTemplate>

		</Router>
	</MsalProvider>

), root)
```

```
16:19:29 [vite] Internal server error: Failed to resolve entry for package "msal-community-solid". The package may have incorrect main/module/exports specified in its package.json.
  Plugin: vite:import-analysis
  File: /home/joshuaring/Projects/my-solid-project/src/index.tsx
      at packageEntryFailure (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23382:11)
      at resolvePackageEntry (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23379:5)
      at tryNodeResolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23113:20)
      at Context.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:22874:28)
      at async Object.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42847:32)
      at async TransformContext.resolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42575:23)
      at async normalizeUrl (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40500:34)
      at async file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40651:47
      at async Promise.all (index 9)
      at async TransformContext.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40577:13)
      at async Object.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42919:30)
      at async loadAndTransform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:53385:29)

```

And when the package is imported but **not used** I get the same error, this time in the browser as well as the terminal running `npm run dev`

I am using SolidJs SPA, Vite, Typscript, this is my `package.json`:

```
{
  "name": "vite-template-solid",
  "version": "0.0.0",
  "description": "",
  "scripts": {
    "start": "vite",
    "dev": "vite",
    "build": "vite build",
    "serve": "vite preview"
  },
  "license": "MIT",
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5.59.7",
    "@typescript-eslint/parser": "^5.59.7",
    "autoprefixer": "^10.4.14",
    "eslint": "^8.41.0",
    "eslint-plugin-solid": "^0.12.1",
    "postcss": "^8.4.23",
    "tailwindcss": "^3.3.2",
    "vite": "^4.1.1",
    "vite-plugin-solid": "^2.5.0"
  },
  "dependencies": {
    "@azure/msal-browser": "^2.37.0",
    "@solidjs/router": "^0.8.2",
    "@tailwindcss/forms": "^0.5.3",
    "chart.js": "^4.3.0",
    "chartjs-adapter-moment": "^1.0.1",
    "chartjs-chart-geo": "^4.2.0",
    "luxon": "^3.3.0",
    "msal-community-solid": "^0.1.3",
    "solid-js": "^1.6.10",
    "typescript": "^5.0.4",
    "vite-plugin-eslint": "^1.8.1"
  }
}
```

**Comment by saraelsa** [comment]
This error is caused by the Solid.js import being specified as `index.jsx` but no such file being generated in `dist` because the entry point is `index.ts`, not `index.tsx`. I've fixed this issue and published the corrected version—please upgrade to version 0.1.4 and confirm if it works properly now.

**Comment by Josh-Ring-SD** [comment]
Slightly better! The package can now can be imported correctly without a crash. But it can not used. 


When you use it, you get:


```
  Plugin: vite-plugin-eslint
  File: /home/joshuaring/Projects/my-solid-project/src/index.tsx
16:52:19 [vite] Internal server error: Failed to resolve entry for package "msal-community-solid". The package may have incorrect main/module/exports specified in its package.json.
  Plugin: vite:import-analysis
  File: /home/joshuaring/Projects/my-solid-project/src/index.tsx
      at packageEntryFailure (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23382:11)
      at resolvePackageEntry (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23379:5)
      at tryNodeResolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23113:20)
      at Context.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:22874:28)
      at async Object.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42847:32)
      at async TransformContext.resolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42575:23)
      at async normalizeUrl (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40500:34)
      at async file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40651:47
      at async Promise.all (index 7)
      at async TransformContext.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40577:13)
      at async Object.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42919:30)
      at async loadAndTransform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:53385:29)
```

**Comment by Josh-Ring-SD** [comment]
Crash reproduced on 0.2.0

```
  File: /home/joshuaring/Projects/my-solid-project/src/index.tsx
16:57:52 [vite] Internal server error: Failed to resolve entry for package "msal-community-solid". The package may have incorrect main/module/exports specified in its package.json.
  Plugin: vite:import-analysis
  File: /home/joshuaring/Projects/my-solid-project/src/index.tsx
      at packageEntryFailure (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23382:11)
      at resolvePackageEntry (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23379:5)
      at tryNodeResolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:23113:20)
      at Context.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:22874:28)
      at async Object.resolveId (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42847:32)
      at async TransformContext.resolve (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42575:23)
      at async normalizeUrl (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40500:34)
      at async file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40651:47
      at async Promise.all (index 9)
      at async TransformContext.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:40577:13)
      at async Object.transform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:42919:30)
      at async loadAndTransform (file:///home/joshuaring/Projects/my-solid-project/node_modules/vite/dist/node/chunks/dep-4d3eff22.js:53385:29)
```

**Comment by saraelsa** [comment]
Should be fixed in 0.2.1 now (as per my testing). There were some `exports` in the `package.json` file pointing to a file that didn't exist which I have removed now.

**Comment by Josh-Ring-SD** [comment]
Nice one, that's brilliant! It's working wonderfully now good job :)

Thanks for your efforts, saved my life
